### ISSUE TITLE
0011386:
Ldap port isn't set automatically anymore

### ISSUE THREAD

**Issue by Gloirin** [tine20/Tine-2.0-Open-Source-Groupware-and-CRM#5634]
**Reported by lab-at-nohl on 18 Oct 2015 20:48**

**Version:** Egon Community Edition (2016.03.1)

I knew this from the forum already but now I stumbled into it myself. Source was fresh git clone from github. 

see details and quick fix:
https://www.tine20.org/forum/viewtopic.php?f=12&amp;t=17566

**Steps to reproduce:** As &#039;laugenbreze&#039; mentioned (forum thread), the error does not occur if url is set with protocol explicitly. 

issue: &quot;localhost&quot;
working: &quot;ldap://localhost&quot;

**Additional information:** I updated target version to latest public version with error.

**Comment by Gloirin** [comment]
**Comment posted by pschuele on 19 Oct 2015 07:45**

this is strange (see my answer in the forum). ldap_connect should use the default port if port = 0. i verified this on my system. could you please check with tcpdump which port is used?

**Comment by Gloirin** [comment]
**Comment posted by lab-at-nohl on 20 Mar 2016 17:01**

I had no luck running &quot;tcpdump dst port 389&quot; - no capture at all. Maybe Zend prefers socket?

**Comment by Gloirin** [comment]
**Comment posted by pschuele on 21 Mar 2016 12:13**

is this still happening in Egon release? we updated the Zend_Ldap classes to 1.12.17, too.

**Comment by Gloirin** [comment]
**Comment posted by lab-at-nohl on 21 Mar 2016 15:23**

YES. Verified yesterday, Egon suffers the same symptoms (I updated target). 

Maybe related to sockets? I tried to change global ldap.conf from ldapi:// to ldap:// but php didn&#039;t respect that. 

Becaus &quot;tcpdump dst port 389&quot; or &quot;tcpdump dst port ldap&quot; was quiet totally I assume there&#039;s no network traffic between php andslapd.
